openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py

```diff
@@ -0,0 +1,635 @@
+    depending upon the data provided in the application.
+    """
+    user_application.resume = resume
+    user_application.cover_letter_file = cover_letter_file
+    user_application.cover_letter = cover_letter_text
+    mock_is_displayable_on_browser.return_value = is_displayable_on_browser
+
+    actual_fields = UserApplicationADGAdmin._get_resume_cover_letter_display_fields('self', user_application)
+
+    assert expected_fields == actual_fields
+
+
+def test_get_fieldset_for_scores():
+    """
+    Test that the `_get_fieldset_for_scores` method returns the correct fieldset.
+    """
+    expected_fieldset = (SCORES, {'fields': (PREREQUISITES,)})
+    actual_fieldset = UserApplicationADGAdmin._get_fieldset_for_scores('self')
+
+    assert expected_fieldset == actual_fieldset
+
+
+@pytest.mark.parametrize(
+    'resume, has_work_experience', [
+        (None, False),
+        (None, True),
+        (TEST_RESUME, True)
+    ], ids=['no_resume, no_experience', 'no_resume, with_experience', 'with_resume']
+)
+@pytest.mark.django_db
+@mock.patch('openedx.adg.lms.applications.admin.ApplicationReviewInline.get_formset')
+@mock.patch('openedx.adg.lms.applications.admin.UserApplicationADGAdmin.get_inline_instances')
+def test_get_formsets_with_inlines(
+    mock_get_inline_instances,
+    mock_get_formset,
+    user_application,
+    user_application_adg_admin_instance,
+    resume,
+    has_work_experience
+):
+    """
+    Test that the overridden generator function `get_formsets_with_inlines`:
+        1. Yields no formsets in case of a user application with attached resume
+        2. Yields formsets for both education and work experience in case the applicant has provided work experience
+        3. Yields formset for only education in case of no work experience.
+    """
+    education_inline_instance = EducationInline(UserApplication, adg_admin_site)
+    work_experience_inline_instance = WorkExperienceInline(UserApplication, adg_admin_site)
+
+    mock_get_inline_instances.return_value = [education_inline_instance, work_experience_inline_instance]
+    mock_get_formset.return_value = 'test_formset'
+
+    user_application.resume = resume
+
+    if has_work_experience:
+        work_experience = WorkExperienceFactory()
+        work_experience.user_application = user_application
+        work_experience.save()
+
+    actual_formsets = UserApplicationADGAdmin.get_formsets_with_inlines(
+        user_application_adg_admin_instance, 'request', user_application
+    )
+
+    if resume:
+        with pytest.raises(StopIteration):
+            next(actual_formsets)
+    else:
+        assert next(actual_formsets) == ('test_formset', education_inline_instance)
+        if has_work_experience:
+            assert next(actual_formsets) == ('test_formset', work_experience_inline_instance)
+        else:
+            with pytest.raises(StopIteration):
+                next(actual_formsets)
```

Comment:
Same is the case with this one, where we have to implement conditions to handle different parameters, we should consider moving them to separate test case.